Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs

```diff
@@ -26,80 +32,146 @@ public class GvrSceneManager : SingletonBehaviour<GvrSceneManager>
+        private void FindSpot(Scene worldScene)
         {
-            if (scene == generalScene)
+            var spots = GameObject.FindGameObjectsWithTag("PxVob_zCVobSpot");
+            for (int i = 0; i < spots.Length; i++)
             {
-                var playerParent = generalScene.GetRootGameObjects().FirstOrDefault(o => o.name == "PlayerController");
-                player = playerParent.transform.Find("VRPlayer_v4 (romey)").gameObject;
-                return;
+                if (spots[i].name == startVobAfterLoading)
+                {
+                    startPoint = spots[i];
+                }
             }
-            SceneManager.SetActiveScene(scene);
-        }
-
-        /// <summary>
-        /// Subscribe the SetActiveScene method so wen can properly place the player in the correct spot.
-        /// </summary>
-        private void ActiveSceneChanged(Scene oldScene, Scene newScene)
-        {
-            if (newScene == GameData.I.WorldScene)
+            if (startPoint == null)
             {
-                // we set manually the XR Interaction Manager as Unity creates a new one in Bootstrap 
-                // and we don't want to use it since we have one in General scene
-                WorldCreator.I.PostCreate(interactionManager.GetComponent<XRInteractionManager>());
-                GameObject startPoint = null;
-                var spots = GameObject.FindGameObjectsWithTag("PxVob_zCVobSpot");
                 for (int i = 0; i < spots.Length; i++)
                 {
-                    if (spots[i].name == startVobAfterLoading)
+                    if ((spots[i].name == "START" || spots[i].name == "START_GOTHIC2") && spots[i].scene == worldScene)
                     {
                         startPoint = spots[i];
                     }
                 }
-                if (startPoint == null)
+            }
+        }
+
+        private async Task ShowLoadingScene(string worldName = null)
+        {
+            generalScene = SceneManager.GetSceneByName(generalSceneName);
+            if (generalScene.isLoaded)
+            {
+                SceneManager.MoveGameObjectToScene(interactionManager, SceneManager.GetSceneByName("Bootstrap"));
+                SceneManager.UnloadSceneAsync(generalScene);
+                generalSceneLoaded = false;
+            }
+
+            // set the loading background texture properly
+            if (worldName != null)
+            {
+            // TODO: for new game we need to load texture "LOADING.TGA"
+                var textureString = "LOADING_" + worldName.Split('.')[0].ToUpper() + ".TGA";
+                UIManager.I.setTexture(textureString, UIManager.I.GothicLoadingMenuMaterial);
+            }
+
+            var loadingScene = SceneManager.LoadScene("Loading", new LoadSceneParameters(LoadSceneMode.Additive));
+
+            SceneManager.sceneLoaded += (Scene scene, LoadSceneMode mode) =>
+            {
+                if (scene == loadingScene)
```

Comment:
Ah I see.

To be honest I don't really like these inline events. They can make it hard to read. Please check if it would make sense to have separate methods subscribed. Then we can easily spot what is called and when.